File: 백준/Silver/1065. 한수/한수.java
import java.util.Scanner;
class Main{
	public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();
        int count = 0;
        for(int i = 1;i<=n;i++){
            if(i<100){
                count+=1;
            }else{
                String num=Integer.toString(i);
                int check = 0;
                for(int j=0;j<num.length()-2;j++){
                    if((num.charAt(j+1)-num.charAt(j))==(num.charAt(j+2)-num.charAt(j+1))){
                        check+=1;
                    }else {
                        break;
                    }
                }
                if(check == num.length()-2){
                    count+=1;
                }
            }
        }
		System.out.println(count);
	}
}

File: 백준/Silver/1065. 한수/한수.py
n = int(input())
count = 0
for i in range(1,n+1):
    if(i<100):
        count+=1
    else:
        num = str(i)
        check = 0
        for j in range(0,len(num)-2):
            if(int(num[j+1])-int(num[j])==int(num[j+2])-int(num[j+1])):
                check+=1
            else:
                break 
        if(check == len(num)-2):
            count+=1
print(count)
    
    
